core/src/main/java/hudson/tasks/LogRotatorPeriodicTask.java

```diff
@@ -0,0 +1,183 @@
+                                        	SEVERE,
+                                        	Messages.LogRotatorPeriodicTask_hasRotator_unknownPolicy(
+                                        		job.getName(),
+                                        		config.getPolicyForJobsWithCustomLogRotator() ) );
+                                        break;
+                                    }
+                                } else {
+                                    // the job has NOT defined its own LogRotator                                    
+                                    switch( config.getPolicyForJobsWithoutCustomLogRotator() ) {
+                                    case NONE:
+                                        LOGGER.log( FINER, Messages.LogRotatorPeriodicTask_noRotator_policyNONE( job.getFullName() ) );
+                                        break;
+                                    case GLOBAL:
+                                        LOGGER.log( FINER, Messages.LogRotatorPeriodicTask_noRotator_policyGLOBAL( job.getFullName() ) );
+                                        applyGlobalLogRotators( job, config.getGlobalLogRotators() );
+                                        break;
+                                    default:
+                                        LOGGER.log(
+                                            SEVERE,
+                                            Messages.LogRotatorPeriodicTask_noRotator_unknownPolicy(
+                                            	job.getName(),
+                                            	config.getPolicyForJobsWithCustomLogRotator() ) );
+                                        break;
+                                    }
+                                }
+                            } catch( InterruptedException ie ) {
+                                // re-throw InterruptedException always, so we're interruptable.
+                                throw ie;
+                            } catch( Exception e ) {
+                                // any other exception, chomp it and try to rotate the rest of the logs.
+                                LOGGER.log(
+                                    WARNING,
+                                    Messages.LogRotatorPeriodicTask_rotationError( job.getFullName() ),
+                                    e );
+                            }
+                        }
+                    }
+                    
+                    config.setLastRotated( System.currentTimeMillis() );
+                }
+            }
+        }
+    }
+    
+    /*
+     * Non-javadoc: will run every minute, but only to check if sufficient time has passed to execute.
+     * Since AsyncPeriodicWork's timer is non-configurable, we need to run every minute, and check
+     * if there's actually anything to do inside the execute(...) 
+     */
+    @Override
+    public long getRecurrencePeriod() { return MIN; }
```

Comment:
That was one of my uncertainties. In the interest of meeting folks' expectations as completely and consistently as possible, `MIN` is a good choice: the _most_ that a log rotation period (defined in hours) could "miss" the exact mark is 59s, or 1/60th (ish) of the rotation period.

However, if there's tolerance for some imprecision, I would say that this should _absolutely_ be HOUR.

You can imagine the situation in which someone saves a configuration to rotate logs hourly 1 minute since the last check. 59 minutes later, it checks again but 1 hour has not passed, and then an hour later it checks and does execute... and the user perceives a 2 hour wait, nearly a 100% difference from the expected period.

I believe HOUR is _fine_ here, but I chose to err on the side of precision for the initial PR.

Reply:
I also think 1 hour is fine. You described the worst scenario but the average one should not be that bad. Also log rotation seems to me like something to think about in long-term, so shouldn't be worrying if one execution is missed.